### DISCUSSION TITLE
Is there a way to more-efficiently model an x-ray vault than a sequence of egs_genvelopes from egs_boxes?

### DISCUSSION THREAD

**Discussion by jburz2001** [nrc-cnrc/EGSnrc#1141] [Questions]
Hello. I need to model the behavior of an x-ray source in an x-ray vault of some geometry. I am starting out by considering this vault to be cubical. Essentially, I have an x-ray source surrounded by air which is then enclosed in a 3 layer wall sandwich of steel-lead-steel, all of which is enveloped by more air. I have constructed a basic model of the geometry using both egs_box and egs_genvelope but I suspect that there is a better way to do it. How would one adjust the attached code for greater computational efficiency / easier implementation?
![image](https://github.com/nrc-cnrc/EGSnrc/assets/56214083/beb81fce-e9d4-489f-a1c8-c7940dddc558)
![image](https://github.com/nrc-cnrc/EGSnrc/assets/56214083/60df69a4-ccaa-4a95-b022-40eac349dbca)

[vault_geometry.egsinp.zip](https://github.com/user-attachments/files/15537940/vault_geometry.egsinp.zip)

**Comment by ftessier** [comment]
I often use a single XYZ geometry to represent all the (rectilinear) geometry. Not necessarily simpler or more efficient, but the expectation is that it would be more efficient since there is a single geometry. 

You would have to see for yourself if you find it easier to code and whether or not it is more efficient. You might want to work up from a simple xyz to define a single box, to familiarize yourself with the region indexing in an xyz.

For example, a 3x3x3 xyz would contain 27 regions. If you set the middle one to air, you  would get a cubical shell!

**Reply by jburz2001** [reply]
Thanks for the tip regarding the XYZGeometry! I'll learn more about how it works and try implementing it to compare the performance. I need to be able to fine-tune the width of the lead portion of the wall so this may prove easier or more difficult for me but I'll find out.
